feat(.NET): CRONS (#9345)
getsentry/sentry-docs: docs/platforms/dotnet/common/crons/hangfire/index.mdx

---
title: Hangfire
description: "Learn more about how to monitor your Hangfire jobs."
sidebar_order: 5001
---

> The .NET SDK provides an integration for [Hangfire](https://www.hangfire.io/) to monitor your jobs by automatically creating check-ins for them. The SDK relies on job filters that get set up when you `UseSenty` like so

Reviewer: ```suggestion
The .NET SDK provides an integration with [Hangfire](https://www.hangfire.io/) to monitor your jobs by automatically [creating check-ins for them](https://docs.sentry.io/product/crons/job-monitoring/). The SDK relies on job filters that get set up when you `UseSentry`. For example:
```

Reviewer 2: ```suggestion
The .NET SDK provides an integration with [Hangfire](https://www.hangfire.io/) to monitor your jobs by automatically [creating check-ins for them](https://docs.sentry.io/product/crons/job-monitoring/). The SDK relies on job filters that are set up when you call `UseSentry`. For example:
```

```csharp
using Hangfire;
using Sentry.Hangfire;

GlobalConfiguration.Configuration.UseSentry();
```

> or if you're using the builder pattern

Reviewer 2: ```suggestion
Alternatively, if you're using the builder pattern, your code may look like this:
```

```csharp
using Hangfire;
using Sentry.Hangfire;

var builder = WebApplication.CreateBuilder(args);
builder.Services.AddHangfire(configuration => configuration
    .UseSentry() // <- Add Sentry to automatically send check-ins
);
```

> To let the SDK know which jobs to monitor you need to provide the job with a monitor slug. This can be done by setting the `SentryMonitorSlug` attribute on the job. An example how to set up a fire-and-forget job that is getting monitored by Sentry could look like the following snippet:

Reviewer: I didn't find any mention of `monitor slug` on the [Crons docs](https://docs.sentry.io/product/crons/) itself. Would be nice to have an explainer and a link. cc @gaprl

Reviewer 2: +1

> To let the SDK know which jobs to monitor you need to provide the job with a monitor slug. This can be done by setting the `SentryMonitorSlug` attribute on the job. An example how to set up a fire-and-forget job that is getting monitored by Sentry could look like the following snippet:

Reviewer 2: ```suggestion
To let the SDK know which jobs to monitor, you need to provide the job with a monitor slug. This can be done by setting the `SentryMonitorSlug` attribute on the job. For example, the following snippet shows how to set up a fire-and-forget job that Sentry will monitor:
```

```csharp
using Hangfire;
using Sentry.Hangfire;

GlobalConfiguration.Configuration.UseSentry();

> BackgroundJob.Enqueue<MyJobClass>(job => job.Execute());
> 
> public class MyJobClass
> {
>     [SentryMonitorSlug("my-monitor-slug")]
>     public void Execute()
>     {
>         // Do your work
>     }
> }

Reviewer: How about a more realistic example?

```suggestion
BackgroundJob.Enqueue<PricingUpdateWorker>(job => job.Execute());

public class PricingUpdateWorker
{
    [SentryMonitorSlug("update-pricing")]
    public void Execute()
    {
        // Do your work to update the price
    }
}
```

```

> The SDK will automatically capture a check-in with `CheckInStatus.InProgress` when the job starts and update the check-in with `CheckInStatus.OK` on success. In case of an exception during execution the check-in will be updated as `CheckInStatus.ERROR` instead.

Reviewer 2: ```suggestion
The SDK will automatically capture a check-in with `CheckInStatus.InProgress` when the job starts and update the check-in with `CheckInStatus.OK` on success. In case of an exception during execution, the check-in will be updated as `CheckInStatus.ERROR` instead.
```
